spec/concurrency_spec.rb

```diff
@@ -1,107 +1,110 @@
   end
-  
+
   after :each do
     Object.send(:remove_const, :ConcurrencyTest)
   end
 
-  
   it "should allow specifying which methods should implement the concurrency check" do
-    lambda { ConcurrencyTest.send(:concurrency_safe, :instance_test_method) }.should_not raise_error
-    lambda { ConcurrencyTest.send(:concurrency_safe, :class_test_method) }.should_not raise_error
-    lambda { ConcurrencyTest.send(:concurrency_safe, :both_instance_and_class_test_method) }.should raise_error(Concurrency::AmbiguousMethodException)
-    lambda { ConcurrencyTest.send(:concurrency_safe, :both_instance_and_class_test_method, :type => :instance) }.should_not raise_error
-    lambda { ConcurrencyTest.send(:concurrency_safe, :both_instance_and_class_test_method, :type => :class) }.should_not raise_error
-    lambda { ConcurrencyTest.send(:concurrency_safe, :unknown_method) }.should raise_error(Concurrency::NoMethodException)
+    expect { ConcurrencyTest.send(:concurrency_safe, :instance_test_method) }
+      .to_not raise_error
+    expect { ConcurrencyTest.send(:concurrency_safe, :class_test_method) }
+      .to_not raise_error
+    expect { ConcurrencyTest.send(:concurrency_safe, :both_instance_and_class_test_method) }
+      .to raise_error(Concurrency::AmbiguousMethodException)
+    expect { ConcurrencyTest.send(:concurrency_safe, :both_instance_and_class_test_method, type: :instance) }
+      .to_not raise_error
+    expect { ConcurrencyTest.send(:concurrency_safe, :both_instance_and_class_test_method, type: :class) }
+      .to_not raise_error
+    expect { ConcurrencyTest.send(:concurrency_safe, :unknown_method) }
+      .to raise_error(Concurrency::NoMethodException)
   end
-  
+
   it "should allow identyfying the type of a method" do
-    ConcurrencyTest.send(:method_types, :class_test_method).should eql ['class']
-    ConcurrencyTest.send(:method_types, :instance_test_method).should eql ['instance']
-    ConcurrencyTest.send(:method_types, :both_instance_and_class_test_method).should eql ['class','instance']
-    ConcurrencyTest.send(:method_types, :unknown_method).should be_blank
-    ConcurrencyTest.send(:method_type, :class_test_method).should eql 'class'
-    ConcurrencyTest.send(:method_type, :instance_test_method).should eql 'instance'
-    lambda { ConcurrencyTest.send(:method_type, :both_instance_and_class_test_method) }.should raise_error(Concurrency::AmbiguousMethodException)
-    lambda { ConcurrencyTest.send(:method_type, :unknown_method) }.should raise_error(Concurrency::NoMethodException)
+    expect(ConcurrencyTest.send(:method_types, :class_test_method)).to eq ['class']
+    expect(ConcurrencyTest.send(:method_types, :instance_test_method)).to eq ['instance']
+    expect(ConcurrencyTest.send(:method_types, :both_instance_and_class_test_method)).to eq ['class','instance']
+    expect(ConcurrencyTest.send(:method_types, :unknown_method)).to be_blank
+    expect(ConcurrencyTest.send(:method_type, :class_test_method)).to eq 'class'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.